modules/service/src/main/scala/ch/epfl/bluebrain/nexus/iam/service/Main.scala

```diff
@@ -69,16 +69,35 @@ object Main {
       logger.info("==== Cluster is Live ====")
       implicit val oidcConfig  = appConfig.oidc
       implicit val baseApiUri  = ApiUri(apiUri)
-      implicit val clock                = Clock.systemUTC
+      implicit val clock       = Clock.systemUTC
       val aggregate            = ShardingAggregate("permission", sourcingSettings)(Initial, Acls.next, Acls.eval)
       val acl                  = Acls[Future](aggregate)
       val downStreamAuthClient = DownstreamAuthClient(cl, uicl)
+      val ownRead              = Permissions(Permission.Own, Permission.Read)
 
-      if (appConfig.auth.adminGroups.isEmpty) {
+      if (appConfig.auth.testMode) {
+        val anonymousCaller = CallerCtx(clock, AnonymousUser)
+        logger.warning("""/!\ Test mode is enabled - this is potentially DANGEROUS /!\""")
+        logger.warning("Granting full rights to every user...")
+        acl.fetch(Path./).onComplete {
+          case Success(mapping) =>
+            mapping.get(Anonymous) match {
+              case Some(permissions) if permissions.containsAll(ownRead) =>
+                logger.info("Top-level permissions found for anonymous; nothing to do")
+              case Some(_) =>
+                logger.info("Adding 'own' & 'read' to top-level permissions for anonymous")
+                acl.add(Path./, Anonymous, ownRead)(anonymousCaller)
+              case None =>
+                logger.info("Creating top-level permissions for anonymous")
+                acl.create(Path./, AccessControlList(Anonymous -> ownRead))(anonymousCaller)
+            }
+          case Failure(e) =>
+            logger.error(e, "Unexpected failure while trying to fetch and set top-level permissions")
+        }
+      } else if (appConfig.auth.adminGroups.isEmpty) {
         logger.warning("Empty 'auth.admin-groups' found in app.conf settings")
         logger.warning("Top-level permissions might be missing as a result")
       } else {
-        val ownRead     = Permissions(Permission.Own, Permission.Read)
         val adminCaller = CallerCtx(clock, AuthenticatedUser(Set(AuthenticatedRef(Some(appConfig.oidc.realm)))))
```

Comment:
I wonder if it would be worth removing `Anonymous` permissions on `/` here? On one hand it might be confusing to remove the  permissions , but on the other hand someone disabling the test mode might be surprised that anonymous users can still access their data.
This would also mean that you would never be able to set `Anonymous` permissions on `/` without the test mode being enabled, which is something we may or may not want.
Just some thoughts we might take into consideration in the future.